lib/chat_api/messages.ex

```diff
@@ -50,9 +51,15 @@ defmodule ChatApi.Messages do
 
   @spec create_message(map()) :: {:ok, Message.t()} | {:error, Ecto.Changeset.t()}
   def create_message(attrs \\ %{}) do
-    %Message{}
-    |> Message.changeset(attrs)
-    |> Repo.insert()
+    with {:ok, message} <-
+           %Message{}
+           |> Message.changeset(attrs)
+           |> Repo.insert() do
+      Conversations.mark_activity(message.conversation_id)
```

Comment:
hmmmm I feel a little FUD about including this as a side effect within the `create_message` method... I know it's a bit more repetitive but I slightly prefer invoking the `Conversations.mark_activity` method explicitly after wherever `create_message` is called 🤔 

I wonder if Phoenix has something similar to Rails' `after commit` hook (i.e. a hook that gets called after a new resource is successfully inserted into the db)... know anything like this?

Reply:
> I know it's a bit more repetitive but I slightly prefer invoking the Conversations.mark_activity method explicitly after wherever create_message is called 🤔

Yeah I had this feeling too. Like we're coupling Conversations and Messages, if I call `Conversations.mark_activity` here. I'm down to changing this so that we explicitly call mark_activity every time create_message is called.  

> I wonder if Phoenix has something similar to Rails' after commit hook (i.e. a hook that gets called after a new resource is successfully inserted into the db)... know anything like this?

From what I see from googling so far, doesn't seem like Ecto has that kind of callback. I'm looking at this reddit [thread](https://www.reddit.com/r/elixir/comments/5yo4ui/ecto_lack_of_callbacks_the_pain/)

Reply:
What do you think of having create_message publish an event somewhere. Then it'll get consumed by the app later, then we call `mark_activity`.

Reply:
I like that! right now we have a lot of repetition around callbacks when a new message is created, so setting up an event like that could be really useful for other things as well 👍 

mind explaining a bit how you're planning on implementing it, just so I have a better sense of what to expect?

Reply:
For sure. I'm thinking of using Oban. 
1. Create a worker called `ChatApi.Workers.NewMessageCreatedActions`
2. Every time a message gets called we enqueue a new job to that worker
3. The worker will be responsible for calling the callbacks that gets triggered when a new message is created. For example calling `Conversations.mark_activity`

Reply:
perfect! for now let's just start off with handling the `mark_activity` logic in that worker, we can handle refactoring some of the other logic later :+1:

Reply:
Sounds good :+1: I'll try to get these changes up tomorrow!

Reply:
great 😄

Reply:
pushed a new commit for this :pray: